test/intl402/Temporal/PlainDate/prototype/since/wrapping-at-end-of-month-persian.js

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2025 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindate.prototype.since
+description: Tests balancing of days to months at end of month (Persian calendar)
+includes: [temporalHelpers.js]
+features: [Temporal, Intl.Era-monthcode]
+---*/
+
+const calendar = "persian";
+
+// Difference between end of longer month to end of following shorter month
+{
+  const end = Temporal.PlainDate.from({ year: 1400, monthCode: "M07", day: 30, calendar });
+  for (const largestUnit of ["years", "months"]) {
+    TemporalHelpers.assertDuration(
+      Temporal.PlainDate.from({ year: 1400, monthCode: "M06", day: 30, calendar }).since(end, { largestUnit }),
+      0, -1, 0, 0, 0, 0, 0, 0, 0, 0,
+      `Shahrivar 30th to Mehr 30th is one month (${largestUnit})`
+    );
+    TemporalHelpers.assertDuration(
+      Temporal.PlainDate.from({ year: 1400, monthCode: "M06", day: 31, calendar }).since(end, { largestUnit }),
+      0, 0, 0, -30, 0, 0, 0, 0, 0, 0,
+      `Shahrivar 31st to Mehr 30th is 30 days, not one month (${largestUnit})`
+    );
+  }
+}
+
+// Difference between end of Bahman to end of Esfand
+{
+  const end = Temporal.PlainDate.from({ year: 1400, monthCode: "M12", day: 29, calendar });
+  for (const largestUnit of ["years", "months"]) {
+    TemporalHelpers.assertDuration(
+      Temporal.PlainDate.from({ year: 1400, monthCode: "M11", day: 29, calendar }).since(end, { largestUnit }),
+      0, -1, 0, 0, 0, 0, 0, 0, 0, 0,
+      `Bahman 29th to Esfand 29th is one month (${largestUnit})`
+    );
+  }
```

Comment:
This should have a test case for 30 Bahman to 29 Esfand in a common-year, being 29 days not 1 month.

(Ditto for the other Persian calendar files)